lib/api/core/validation/index.js

```diff
@@ -1,690 +1,682 @@
+    const
+      query = {
+        bool: {
+          must: validatorFilter
+        }
+      };
 
-  promise = this.dsl.validate(query);
+    const promise = this.dsl.validate(query);
 
-  if (!dryRun) {
-    promise.then(() => {
-      return this.dsl.register(indexName, collectionName, query);
-    });
+    if (!dryRun) {
+      promise.then(() => this.dsl.register(indexName, collectionName, query));
+    }
+
+    return promise;
   }
 
-  return promise;
-};
+  /**
+   * @param {ValidationType} validationType
+   * @throws {PluginImplementationError}
+   */
+  addType(validationType) {
+    if (!validationType.typeName) {
+      throw new PluginImplementationError('The typeName property must be defined in the validation type object.');
+    }
 
-/**
- * @param {ValidationType} validationType
- * @throws {PluginImplementationError}
- */
-Validation.prototype.addType = function validationAddType (validationType) {
-  if (!validationType.typeName) {
-    throw new PluginImplementationError('The typeName property must be defined in the validation type object.');
-  }
+    if (!validationType.validate || typeof validationType.validate !== 'function') {
+      throw new PluginImplementationError(`The type ${validationType.typeName} must implement the function 'validate'.`);
+    }
 
-  if (!validationType.validate || typeof validationType.validate !== 'function') {
-    throw new PluginImplementationError(`The type ${validationType.typeName} must implement the function 'validate'.`);
-  }
+    if (!validationType.validateFieldSpecification || typeof validationType.validateFieldSpecification !== 'function') {
+      throw new PluginImplementationError(`The type ${validationType.typeName} must implement the function 'validateFieldSpecification'.`);
+    }
 
-  if (!validationType.validateFieldSpecification || typeof validationType.validateFieldSpecification !== 'function') {
-    throw new PluginImplementationError(`The type ${validationType.typeName} must implement the function 'validateFieldSpecification'.`);
-  }
+    if (validationType.hasOwnProperty('allowChildren') && validationType.allowChildren) {
+      if (!validationType.getStrictness || typeof validationType.getStrictness !== 'function') {
+        throw new PluginImplementationError(`The allowing children type ${validationType.typeName} must implement the function 'getStrictness'.`);
+      }
 
-  if (validationType.hasOwnProperty('allowChildren') && validationType.allowChildren) {
-    if (!validationType.getStrictness || typeof validationType.getStrictness !== 'function') {
-      throw new PluginImplementationError(`The allowing children type ${validationType.typeName} must implement the function 'getStrictness'.`);
+      this.typeAllowsChildren.push(validationType.typeName);
+    }
+    if (this.types[validationType.typeName]) {
+      throw new PluginImplementationError(`The type ${validationType.typeName} is already defined.`);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 101 characters long line (which is greater than 80 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ALineLength)